sequence_processing_pipeline/Job.py

```diff
@@ -290,3 +292,26 @@ def qsub(self, script_path, qsub_parameters=None,
         else:
             # job was never in the queue - return an error.
             raise PipelineError("job %s never appeared in the queue." % job_id)
+
+    def _group_commands(self, cmds):
+        # break list of commands into chunks of max_array_length (Typically
+        # 1000 for Torque job arrays). To ensure job arrays are never more
+        # than 1000 jobs long, we'll chain additional commands together, and
+        # evenly distribute them amongst the first 1000.
+        cmds.sort()
+        chunks = [cmds[i:i + self.max_array_length] for i in
```

Comment:
This strategy won't balance across chunks, not sure if that is important here. We do something similar that does balance in qp-woltka though, mentioning incase relevant

Reply:
Thanks @wasade ! I'll take a look at qp-woltka as an improvement on this implementation.

Reply:
I don't think chunk balancing is important but [the code is here](https://github.com/qiita-spots/qp-woltka/blob/main/qp_woltka/woltka.py#L40-L54) in case you want to take a look.

Reply:
ty!